settings/OpeningPeriodForm/BigCalendarWrapper.js

```diff
@@ -122,7 +125,37 @@ class BigCalendarWrapper extends React.PureComponent {
       this.onCalendarChange(nextEvents);
     };
 
+  checkEventExistOrOverlap = (startTime, endTime) => {
+    const { events } = this.state;
+    const existingEventsArr = events.map(event => {
+      const { start, end } = event;
+      const startOfExistedEvent = moment(start);
+      const endOfExistedEvent = moment(end);
+      const startTimeMoment = moment(startTime);
+      const endTimeMoment = moment(endTime);
+      const dublicatedEvent = startTimeMoment.isSame(startOfExistedEvent) && endTimeMoment.isSame(endOfExistedEvent);
+      const startBeforeEvent = startTimeMoment.isBefore(startOfExistedEvent)
+        && endTimeMoment.isSameOrBefore(endOfExistedEvent)
+        && endTimeMoment.isAfter(startOfExistedEvent);
+      const startAfterEvent = startTimeMoment.isSameOrAfter(startOfExistedEvent)
+        && startTimeMoment.isBefore(endOfExistedEvent)
+        && (endTimeMoment.isSameOrAfter(endOfExistedEvent) || endTimeMoment.isSameOrBefore(endOfExistedEvent));
+      const fullOverlap = (startTimeMoment.isAfter(startOfExistedEvent) && endTimeMoment.isBefore(endOfExistedEvent))
+        || (startTimeMoment.isBefore(startOfExistedEvent) && endTimeMoment.isAfter(endOfExistedEvent));
+
+      if (dublicatedEvent || startBeforeEvent || startAfterEvent || fullOverlap) {
+        return false;
+      }
+      return true;
+    });
+
+    return existingEventsArr.every(event => event);
```

Comment:
```suggestion
    return existingEvents.every(event => event);
```

Try not to mention type in variable names. Types info does not bring much value and may leed to confusion when the type is changed but the name of the variable is not.